presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorMetadata.java

```diff
@@ -965,4 +965,12 @@ default ConnectorTableProperties getTableProperties(ConnectorSession session, Co
      * <p>
      */
     default void validateScan(ConnectorSession session, ConnectorTableHandle handle) {}
+
+    /**
+     * Redirects to another catalog which may or may not use the same connector
+     */
+    default Optional<String> redirectCatalog(ConnectorSession session, SchemaTableName tableName)
```

Comment:
I expect this to return a new FQN.  I know is the cases we are dealing with are all simple "change the catalog" case, but a change to `CatalogSchemTableName` would be much more powerful here and support a semantic catalog.  It is also easy to convert from this current code to a FQN version.

Reply:
We intentionally do not rewrite the schema or table name here, as that makes operations like renames nonsensical. What you are describing is a completely different feature that we considered but rejected in favor of the catalog redirection approach. I agree that we want both, eventually, but the semantics of each need to be carefully considered. The restriction to only allow catalog here actually makes it more powerful in that it can support features like rename.